### PR TITLE
JAMES-2508 TMPFS for RabbitMQ and other improvments

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: server/mailet/integration-testing/src/test/java/org/apache/james/transport/mailets/AmqpForwardAttachmentTest.java [hunk 1]
```diff
@@ -63,7 +65,9 @@
     private static final byte[] TEST_ATTACHMENT_CONTENT = "Test attachment content".getBytes(StandardCharsets.UTF_8);
 
     public SwarmGenericContainer rabbitMqContainer = new SwarmGenericContainer(Images.RABBITMQ)
-            .withAffinityToContainer();
+            .withAffinityToContainer()
```
Comment (mbaechler) [inline server/mailet/integration-testing/src/test/java/org/apache/james/transport/mailets/AmqpForwardAttachmentTest.java]: you should rather create a new Container type for that

**Comment by chibenwa** [comment]
>  how much do we gain ?

I estimate  minute on my machine, but CI is running. I will update when possible.

**Comment by chibenwa** [comment]
Before:

```
[INFO] Apache James :: Server :: Mail Queue :: RabbitMQ 3.2.0-SNAPSHOT SUCCESS [04:03 min]
[INFO] Apache James :: Server :: Mailets Integration Testing SUCCESS [04:58 min]
```

After:

```
[INFO] Apache James :: Server :: Mail Queue :: RabbitMQ 3.2.0-SNAPSHOT SUCCESS [02:55 min]
[INFO] Apache James :: Server :: Mailets Integration Testing SUCCESS [04:54 min]
```

Same host **obiwan**

Gain: **1min12s**

I bet I can exlude the changes in **Mailets Integration Testing** from this PR...

File: server/mailet/integration-testing/src/test/java/org/apache/james/transport/mailets/AmqpForwardAttachmentTest.java (hunk 1 above)
Comment (chibenwa) [reply server/mailet/integration-testing/src/test/java/org/apache/james/transport/mailets/AmqpForwardAttachmentTest.java]: This commit is removed

**Comment by chibenwa** [comment]
```
[9b3a7e169d1520324787c698839c5f9265082fd6] Failed tests: 
[9b3a7e169d1520324787c698839c5f9265082fd6]   DistantMailboxPathRegisterTest.unregisterMixedWithRegisterShouldWorkInAConcurrentEnvironment:328 
[9b3a7e169d1520324787c698839c5f9265082fd6] Expecting:
[9b3a7e169d1520324787c698839c5f9265082fd6]  <{namespace:user:name=200L}>
[9b3a7e169d1520324787c698839c5f9265082fd6] to contain:
[9b3a7e169d1520324787c698839c5f9265082fd6]  <[MapEntry[key=namespace:user:name, value=100L]]>
[9b3a7e169d1520324787c698839c5f9265082fd6] but could not find:
[9b3a7e169d1520324787c698839c5f9265082fd6]  <[MapEntry[key=namespace:user:name, value=100L]]>
[9b3a7e169d1520324787c698839c5f9265082fd6] 
```

test this please

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**